modules/services/sketchybar/default.nix

```diff
@@ -0,0 +1,66 @@
+{ config, lib, pkgs, ... }:
+
+let
+  inherit (lib) literalExpression maintainers mdDoc mkEnableOption mkIf mkPackageOptionMD mkOption types;
+
+  cfg = config.services.sketchybar;
+
+  configHome = pkgs.writeTextFile {
+    name = "sketchybarrc";
+    text = cfg.config;
+    destination = "/sketchybar/sketchybarrc";
+    executable = true;
+  };
+in
+
+{
+
+  meta.maintainers = [
+    maintainers.azuwis or "azuwis"
+  ];
+
+  options.services.sketchybar = {
+    enable = mkEnableOption (mdDoc "sketchybar");
+
+    package = mkPackageOptionMD pkgs "sketchybar" { };
+
+    extraPackages = mkOption {
+      type = types.listOf types.package;
+      default = [ ];
+      description = mdDoc ''
+        Extra packages to add to PATH.
+      '';
+      example = literalExpression "[ pkgs.jq ]";
+    };
+
+    config = mkOption {
+      type = types.lines;
+      default = "";
+      example = literalExpression ''
+        sketchybar --bar height=24
+        sketchybar --update
+        echo "sketchybar configuration loaded.."
+      '';
+      description = mdDoc ''
+        Content of configuration file. If empty(default), configuration file won't be managed.
+
+        See [documentation](https://felixkratz.github.io/SketchyBar/)
+        and [example](https://github.com/FelixKratz/SketchyBar/blob/master/sketchybarrc).
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+
+    launchd.user.agents.sketchybar = {
+      path = [ cfg.package ] ++ cfg.extraPackages ++ [ config.environment.systemPath ];
+      serviceConfig.ProgramArguments = [ "${cfg.package}/bin/sketchybar" ];
+      serviceConfig.KeepAlive = true;
+      serviceConfig.RunAtLoad = true;
+      serviceConfig.EnvironmentVariables = {
+        XDG_CONFIG_HOME = mkIf (cfg.config != "") "${configHome}";
+      };
```

Comment:
This feels like it should live in a `home-manager` module

Reply:
Last time I checked, sketchybar does not provide command line arg to change config file path, but use XDG_CONFIG_HOME env.

Now it have `--config` option, I'll update the module.